packages/gallery/src/components/item/videos/videoItem.js

```diff
@@ -81,13 +97,60 @@ class VideoItem extends React.Component {
     return this.isHLSVideo() && utils.isiOS();
   }
 
+  updatePlayer = async () => {
+    if (!this.isYoutubeVideo()) {
+      return;
+    }
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+      // if the user click play on iframe and state not yet updated break
+      if (this.state.shouldPlay && this.playingFromIframeClick) {
+        this.playingFromIframeClick = false;
+      }
+      if (!this.state.shouldPlay && this.pausingFromIframeClick) {
+        this.pausingFromIframeClick = false;
+      }
+      if (!this.state.shouldPlay && this.playingFromIframeClick) {
+        break;
+      }
+      if (this.state.shouldPlay && this.pausingFromIframeClick) {
+        break;
+      }
+      if (this.video?.player?.player?.player?.playVideo) {
```

Comment:
All these conditions won't be needed in case this code runs for youtube only. Correct?